.circleci/config.yml

```diff
@@ -125,6 +139,132 @@ jobs:
+
+# TODO: add after fixing the python issue regarding missing bdist_wheel
+#  test-assembly-windows-zip:
+#    machine: true
+#    working_directory: ~/grakn
+#    steps:
+#      - checkout
+#      - run:
+#          command: .circleci/test/assembly/windows-zip/windows-zip.py # TODO: test starting Grakn from a directory which has space
+#          no_output_timeout: 20m
+
+  test-assembly-linux-targz:
+    machine: true
+    working_directory: ~/grakn
+    steps:
+      - checkout
+      - install-bazel
+      - run:
+          command: |
+            bazel build //:assemble-linux-targz
+            tar -xf bazel-genfiles/grakn-core-all-linux.tar.gz -C bazel-genfiles
+            nohup bazel-genfiles/grakn-core-all-linux/grakn server start
+      - run: bazel test //test-deployment:test-deployment --test_output=streamed --spawn_strategy=standalone --cache_test_results=no
+      - run: bazel-genfiles/grakn-core-all-linux/grakn server stop
+
+  test-assembly-apt:
+    machine: true
+    working_directory: ~/grakn
+    steps:
+      - checkout
+      - install-bazel
+      - run:
+          command: |
+            echo $(date +%s)-$(cat VERSION)-$CIRCLE_SHA1 > VERSION
+            echo VERSION=$(cat VERSION)
+      - run: bazel build //bin:assemble-apt
+      - run: bazel build //server:assemble-apt
+      - run: bazel build //console:assemble-apt
+      - run: sudo dpkg -i bazel-bin/bin/grakn-core-bin__all.deb
+      - run: sudo dpkg -i bazel-bin/server/grakn-core-server__all.deb
+      - run: sudo dpkg -i bazel-bin/console/grakn-core-console__all.deb
+      - run: sudo chown -R circleci:circleci /opt/grakn/ # TODO: how do we avoid having to chown?
+      - run: nohup grakn server start
+      - run: bazel test //test-deployment:test-deployment --test_output=streamed --spawn_strategy=standalone --cache_test_results=no
+      - run: grakn server stop
+      # TODO: how to get repo.grakn.ai apt repository to not ignore commit id in the version? that way we don't have to add date +%s
+      - run: bazel run //bin:deploy-apt -- test $REPO_GRAKN_USERNAME $REPO_GRAKN_PASSWORD
+      - run: bazel run //console:deploy-apt -- test $REPO_GRAKN_USERNAME $REPO_GRAKN_PASSWORD
+      - run: bazel run //server:deploy-apt -- test $REPO_GRAKN_USERNAME $REPO_GRAKN_PASSWORD
+      - persist_to_workspace:
+          root: ~/grakn
+          paths:
+            - VERSION
+
+  test-deployment-apt:
+    machine: true
+    working_directory: ~/grakn
+    steps:
+      - checkout
+      - attach_workspace:
+          at: ~/circleci-workspace
+      - install-bazel
+      - run: echo "deb [ arch=all ] https://repo.grakn.ai/repository/test-deb/ trusty main" | sudo tee -a /etc/apt/sources.list.d/grakn-core.list
+      - run: sudo apt-key adv --keyserver keyserver.ubuntu.com --recv 8F3DA4B5E9AEF44C
+      - run: sudo apt update
+      - run:
+          command: |
+            mv ~/circleci-workspace/VERSION VERSION
+            cat VERSION
```

Comment:
I find it easier to read the config files when we just write these commands like this, ganesh:
```
- run: mv ~/circleci-workspace/VERSION VERSION
- run: cat VERSION
```
yes, it creates more "CI" steps in the web browser, but it makes the config file more "aligned" and "uniform".

Reply:
fixed in 39c0b57d6e8d9e7b2e83d70ccfeaaf5e70757609

Reply:
Actually, even `cat VERSION` doesn't need a `name:`, Ganesh. It's still a very basic bash command that it doesn't need to be too verbose.

Reply:
fixed in f7c3ce3daa038fe4dc7aff7f125f620e64e3dfa9

Reply:
there's another in `test-assembly-rpm` @lolski

Reply:
Whoops! Fixed ccf8b3ec1019daf7ec8bf6d9a0258786868a43b4